pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1553,3 +1557,21 @@ func protoEncodeFlowlogCounters(counters types.FlowlogCounters) *metrics.Flowlog
 		FailedAttempts: counters.FailedAttempts,
 	}
 }
+
+// getDormantTime returns scaled dormant time
+func getDormantTime(ctx *zedagentContext) uint64 {
+	return uint64(ctx.globalConfig.GlobalValueInt(types.MetricInterval) * ctx.dormantTimeScaleFactor)
+}
+
+// scaleDormantTimeScaleFactor scale's dormantTimeScaleFactor up or down (depending upon scaleUp)
+// by metricsFailureScaleFactor. Updated dormantTimeScaleFactor should  always be
+// >= minDormantTimeScaleFactor && <= maxDormantTimeScaleFactor
+func scaleDormantTimeScaleFactor(ctx *zedagentContext, scaleUp bool) {
+	if scaleUp {
+		ctx.dormantTimeScaleFactor = uint32(math.Ceil(
+			math.Min(float64(ctx.dormantTimeScaleFactor*metricsFailureScaleFactor), maxDormantTimeScaleFactor)))
+	} else {
+		ctx.dormantTimeScaleFactor = uint32(math.Ceil(
```

Comment:
If we are going to have this dynamic adjustment I think it should be modeled on the TCP CWND AIMD which means that when there is a failure we double the factor, but when there is a success we subtract one from the factor.

As I said in another comment, I think the current min factor of three should remain, but I don't have a good suggestion for a max. Let's stick with 8 as you have now.

Reply:
Noted, will be addressed in the PR for dynamic adjustment.